react-app/src/components/property/PropertyDialog.tsx

```diff
@@ -324,18 +430,75 @@ export const PropertyNetBookValueEditDialog = (props: IPropertyNetBookValueEditD
   );
   const netBookFormMethods = useForm({
     defaultValues: { Fiscals: [] },
+    mode: 'onBlur',
   });
+  const currentYear = new Date().getFullYear();
+  const years = [currentYear];
+  const defaultValues = years.map((year) => ({
+    FiscalYear: year,
+    Value: 0,
+    EffectiveDate: undefined,
+    FiscalKeyId: 0,
+    PropertyType: initialValues?.PropertyTypeId || null,
+    Id: initialValues?.Id || null,
+  }));
 
   useEffect(() => {
-    netBookFormMethods.reset({
-      Fiscals: initialValues?.Fiscals?.map((fisc) => ({
+    if (!initialValues?.Fiscals || initialValues.Fiscals.length === 0) {
+      // use default values if there are no initial values for fiscal years
+      netBookFormMethods.reset({ Fiscals: defaultValues });
+    } else {
+      const fiscalYears =
+        initialValues?.Fiscals?.map((fiscal: { FiscalYear: any }) => fiscal.FiscalYear) ?? [];
+
+      const fiscalValues = initialValues.Fiscals.map((fisc) => ({
         ...fisc,
         Value: String(fisc.Value).replace(/[$,]/g, ''),
         EffectiveDate: dayjs(fisc.EffectiveDate),
-      })),
-    });
+      }));
+      // Check if currentYear is not in yearsFromEvaluations array
+      if (!fiscalYears.includes(currentYear)) {
+        // Add currentYear to yearsFromEvaluations array
+        fiscalYears.unshift(currentYear);
```

Comment:
I don't think this is used again later, so is it necessary to add the current year?